retype.py

```diff
@@ -169,7 +169,13 @@ def lib2to3_parse(src_txt):
         result = drv.parse_string(src_txt, True)
     except ParseError as pe:
         lineno, column = pe.context[1]
-        faulty_line = src_txt.splitlines()[lineno - 1]
+        lines = src_txt.splitlines()
+        if lineno == len(lines) + 1 and column == 0:
+            faulty_line = "Perhaps the source is missing a trailing newline."
```

Comment:
You don't have to guess. Just check if `src_txt[-1]` is a newline. If it is, report the error accordingly. If it **does** end with a newline, returning the last line available instead is probably okay.